phpBB/includes/functions.php

```diff
@@ -2745,7 +2745,8 @@ function get_preg_expression($mode)
 			return array(
 				'#<!\-\- e \-\-><a href="mailto:(.*?)">.*?</a><!\-\- e \-\->#',
 				'#<!\-\- l \-\-><a (?:class="[\w-]+" )?href="(.*?)(?:(&amp;|\?)sid=[0-9a-f]{32})?">.*?</a><!\-\- l \-\->#',
-				'#<!\-\- ([mw]) \-\-><a (?:class="[\w-]+" )?href="(.*?)">(.*?)</a><!\-\- \1 \-\->#',
+				'#<!\-\- ([mw]) \-\-><a (?:class="[\w-]+" )?href="http://(.*?)">\2</a><!\-\- \1 \-\->#',
```

Comment:
I get the feeling that you don't exactly know what this part actually does. The regex is used in the decode_message() function. This means it's only used to decode messages that were saved in the database prior to 3.2. In addition to that, it only matches URLs that are automatically parsed as URLs (magic URLs) and that have automatically been prefixed with the `http://` prefix by phpBB (which can be assumed by the `http://` prefix missing in the URL title.

Why we're not getting rid of prefixes? The scheme is obligatory in the href of hyperlinks. They should not be omitted (although some browsers do not care about the scheme missing).
Prefixing with `http://` is something that should always work if the scheme is not known. Servers should properly take care of redirecting to e.g. `https://` if that is the preferred scheme.

If you post bar.phpbb.com you didn't specify a prefix. One can assume that it's supposed to be http as that is the most widely used scheme. If you really mean any non standard scheme then you should specify that.